spec/src/main/asciidoc/chapters/introduction/entities.asciidoc

```diff
@@ -0,0 +1,142 @@
+Every atomic value has a named type, referred to in the sister-specifications of this specification as a _basic type_.
+
+[NOTE]
+The atomic value types allowed by an implementation of Jakarta Query depend on the programming language and underlying database technology.
+
+Atomic values of distinct type are never considered identical (equal).
+Whether two atomic values of the same type are considered identical depends on semantics which are specific to the type.
+For example, two Unicode strings are identical if they have the same length and the same Unicode character at each position.
+Identity for a given atomic value type must be reflexive, symmetric, and transitive.
+
+=== Collections
+
+A _collection_ is a finite set, list, or map containing atomic values or structures.
+A collection might contain atomic values, or it might contain structures.
+The keys of a map must be atomic values.
+
+Collections are homogeneous.
+A collection may not contain both atomic values and structures.
+
+- If a collection contains atomic values, then every atomic value has the same type.
+- if a collection contains structures, then every structure has the same type.
+- Every key of a map has the same atomic value type.
+
+A collection therefore has a type derived from the type of the atomic values or structures it contains--and, for a map, from the type of its keys.
+We consider that two collections have the same type if they contain the same type of atomic value or structure.
+For a map, we also require that they have the same type of key.
+
+A collection has no identity independent of the values it contains.
+
+- Two sets are identical if they contain exactly the same elements.
+- Two lists are identical if they contain exactly the same elements at exactly the same positions.
+- Two maps are identical if they contain exactly the same keys, and identical values for each key.
+
+=== Structures and records
+
+A _structure_ is a finite set of labelled elements, each of which is an atomic value, a nested structure, or a collection.
+
+Structures are inhomogeneous.
+Elements with distinct labels may have distinct types.
+
+Every structure has a type derived from the types of its elements.
+We consider that two structures have the same type if:
+
+- they contain the same labels, and
+- for each label, the labelled elements have the same type.
+
+A structure has no identity independent of the labelled elements it contains.
+Two structures are identical if they contain exactly the same labels, and identical atomic values, collections, or nested structures for each label.
+
+For example--borrowing a convenient notation--the following structure might represent a book:
+
+[source]
+{
+  isbn: "1-85723-235-6",
+  title: "Feersum Endjinn",
+  pages: 279,
+  year: 1994,
+  authors: {
+    {
+      name: "Iain M. Banks"
+    }
+  }
```

Comment:
This might help to convey that authors is a collection rather than a structure,
```suggestion
  authors: [
    {
      name: "Iain M. Banks"
    }
  ]
```

Reply:
Good idea. I used braces because I figured it was a set, but in fact it should be a list anyway.